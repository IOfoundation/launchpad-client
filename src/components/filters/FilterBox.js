import React from 'react';
import {PropTypes} from 'prop-types';
import FilterByText from './FilterByText';
import FilterByOptions from './FilterByOptions';

class FilterBox extends React.Component {
  render() {
    return (
      <div className="container-invert">
        <div className="grid">
          <FilterByText
            filterName={'Search Box'}
            handleTextSearchBusinesses={this.props.handleTextSearchBusinesses}
            handleOnRemoveFilterOption={this.props.handleOnRemoveFilterOption}
            handleClickOnClearAllFilters={
              this.props.handleClickOnClearAllFilters
            }
            getFilterChips={this.props.getFilterChips}
            search_results={this.props.search_results}
          />
        </div>
        <div className="grid container--middle bottom-xs between-xs p-left-0">
          <FilterByOptions
            filterName={'Business Services'}
            filterOptions={this.props.filterOptions.businessServices}
            handleOnChangeFilterOptions={this.props.handleOnChangeFilterOptions}
          />
          <FilterByOptions
            filterName={'Industry'}
            filterOptions={this.props.filterOptions.industries}
            handleOnChangeFilterOptions={this.props.handleOnChangeFilterOptions}
          />
          <FilterByOptions
            filterName={'Business Stage'}
            filterOptions={this.props.filterOptions.stages}
            handleOnChangeFilterOptions={this.props.handleOnChangeFilterOptions}
          />
          <FilterByOptions
            filterName={'Underserved Com...'}
            filterOptions={this.props.filterOptions.communities}
            handleOnChangeFilterOptions={this.props.handleOnChangeFilterOptions}
          />
        </div>
      </div>
    );
  }
}

FilterBox.propTypes = {
  filterOptions: PropTypes.object.isRequired,
  getFilterChips: PropTypes.func.isRequired,
  handleClickOnClearAllFilters: PropTypes.func.isRequired,
  handleOnChangeFilterOptions: PropTypes.func.isRequired,
  handleOnRemoveFilterOption: PropTypes.func.isRequired,
  handleTextSearchBusinesses: PropTypes.func.isRequired,
  search_results: PropTypes.arrayOf(PropTypes.object),
};

export default FilterBox;
