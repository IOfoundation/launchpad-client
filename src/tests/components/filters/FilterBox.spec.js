import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import FilterBox from '../../../components/filters/FilterBox.js';
import organizationFixture from '../../fixtures/organization';
import filtersFixture from '../../fixtures/filters';

const createProps = props => {
  return {
    businesses: {
      appliedFilters: props.appliedFilters,
      filters: filtersFixture,
      items: props.items,
      organizations: [organizationFixture],
    },
    filterById: false,
    getTextSearchResults: jest.fn(),
    handleClickOnClearAllFilters: jest.fn(),
    handleOnChangeFilterOptions: jest.fn(),
  };
};

describe('<FilterBox />', () => {
  it('renders snapshot of FilterBox', () => {
    const props = createProps({
      appliedFilters: {page: 1},
      items: [],
    });
    const wrapper = shallow(<FilterBox {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
