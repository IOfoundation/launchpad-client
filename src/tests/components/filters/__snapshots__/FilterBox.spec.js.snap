// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterBox /> renders snapshot of FilterBox 1`] = `
<div
  className="container-invert"
>
  <div
    className="grid"
  >
    <OnClickOutside(FilterByText)
      appliedFilters={
        Object {
          "page": 1,
        }
      }
      eventTypes={
        Array [
          "mousedown",
          "touchstart",
        ]
      }
      excludeScrollbar={false}
      filterById={false}
      filterName="Search Box"
      getTextSearchResults={[Function]}
      handleClickOnClearAllFilters={[Function]}
      handleOnChangeFilterOptions={[Function]}
      items={Array []}
      organizations={
        Array [
          Object {
            "contacts": Array [
              Object {
                "email": "jondoe@test.org",
                "id": 1,
                "name": "Jon Doe",
                "phones": Array [
                  Object {
                    "number": "(530) 795-1520",
                  },
                ],
              },
            ],
            "description": "orgDescription",
            "email": "jondoe@something.org",
            "id": 1,
            "locations": Array [
              Object {
                "address": Object {
                  "address_1": "123 Road St",
                  "address_2": null,
                  "city": "Sacramento",
                  "postal_code": "12345",
                  "state_province": "CA",
                },
                "id": 1,
                "name": "location1",
                "phones": Array [
                  Object {
                    "id": 2,
                    "number": "(123) 456 - 7890",
                  },
                ],
              },
              Object {
                "address": Object {
                  "address_1": "456 Street Rd",
                  "address_2": null,
                  "city": "Sacramento",
                  "postal_code": "12345",
                  "state_province": "CA",
                },
                "id": 2,
                "name": "location2",
                "phones": Array [
                  Object {
                    "id": 3,
                    "number": "(123) 456 - 7890",
                  },
                ],
              },
            ],
            "name": "organization1",
            "phones": Array [
              Object {
                "id": 1,
                "number": "(123) 456 - 7890",
              },
            ],
            "services": Array [
              Object {
                "categories": Array [
                  Object {
                    "id": 54,
                    "name": "category1",
                  },
                  Object {
                    "id": 55,
                    "name": "category2",
                  },
                  Object {
                    "id": 56,
                    "name": "category3",
                  },
                ],
                "description": "lorem ipsum...",
                "email": null,
                "id": 1,
                "name": "service1",
                "phones": Array [],
              },
              Object {
                "categories": Array [
                  Object {
                    "id": 54,
                    "name": "category1",
                  },
                ],
                "description": "lorem ipsum...",
                "email": null,
                "id": 2,
                "name": "service2",
                "phones": Array [],
              },
            ],
          },
        ]
      }
      outsideClickIgnoreClass="ignore-react-onclickoutside"
      preventDefault={false}
      stopPropagation={false}
    />
  </div>
  <div
    className="grid col-lg-10 col-md-12 col-xs-12 bottom-xs between-xs p-left-0"
  >
    <OnClickOutside(FilterByOptions)
      eventTypes={
        Array [
          "mousedown",
          "touchstart",
        ]
      }
      excludeScrollbar={false}
      filterName="Business Services"
      filterOptions={
        Array [
          Object {
            "children": Array [
              Object {
                "id": 3,
                "name": "Bookkeeping/Accounting Software Training",
              },
              Object {
                "id": 4,
                "name": "Budget, Financial Statement & Cash Flow Training",
              },
            ],
            "id": 2,
            "name": "Financial Management",
          },
          Object {
            "children": Array [
              Object {
                "id": 11,
                "name": "Accelerator Programs",
              },
              Object {
                "id": 12,
                "name": "Equity Investors: Angels",
              },
            ],
            "id": 10,
            "name": "Capital",
          },
        ]
      }
      handleOnChangeFilterOptions={[Function]}
      outsideClickIgnoreClass="ignore-react-onclickoutside"
      preventDefault={false}
      stopPropagation={false}
    />
    <OnClickOutside(FilterByOptions)
      eventTypes={
        Array [
          "mousedown",
          "touchstart",
        ]
      }
      excludeScrollbar={false}
      filterName="Industry"
      filterOptions={
        Array [
          Object {
            "id": 172,
            "name": "Management",
          },
          Object {
            "id": 180,
            "name": "Retail",
          },
        ]
      }
      handleOnChangeFilterOptions={[Function]}
      outsideClickIgnoreClass="ignore-react-onclickoutside"
      preventDefault={false}
      stopPropagation={false}
    />
    <OnClickOutside(FilterByOptions)
      eventTypes={
        Array [
          "mousedown",
          "touchstart",
        ]
      }
      excludeScrollbar={false}
      filterName="Business Stage"
      filterOptions={
        Array [
          Object {
            "id": 146,
            "name": "Idea/Inception",
          },
          Object {
            "id": 147,
            "name": "Proof of Concept/Prototype/Market Intro",
          },
        ]
      }
      handleOnChangeFilterOptions={[Function]}
      outsideClickIgnoreClass="ignore-react-onclickoutside"
      preventDefault={false}
      stopPropagation={false}
    />
    <OnClickOutside(FilterByOptions)
      eventTypes={
        Array [
          "mousedown",
          "touchstart",
        ]
      }
      excludeScrollbar={false}
      filterName="Underserved Com..."
      filterOptions={
        Array [
          Object {
            "id": 151,
            "name": "African American",
          },
          Object {
            "id": 156,
            "name": "Native American",
          },
        ]
      }
      handleOnChangeFilterOptions={[Function]}
      outsideClickIgnoreClass="ignore-react-onclickoutside"
      preventDefault={false}
      stopPropagation={false}
    />
  </div>
</div>
`;
